core/src/code.rs

```diff
@@ -398,10 +418,20 @@ impl Code {
         }
 
         if let Some(stack_limit) = config.stack_height {
+            let globals = config.export_stack_height.then(|| module.globals_space());
+
             module = wasm_instrument::inject_stack_limiter(module, stack_limit).map_err(|err| {
                 log::trace!("Failed to inject stack height limits: {err}");
                 CodeError::StackLimitInjection
             })?;
+
+            if let Some(globals_before) = globals {
+                // ensure stack limiter injector has created global
+                let globals_after = module.globals_space();
+                assert_eq!(globals_after, globals_before + 1);
+
+                module = export_stack_height(module);
```

Comment:
maybe we can move this code to `gwasm-instrument` crate?

Reply:
I'd leave it here to avoid many patches in our fork